### ISSUE TITLE
My code doesn't work. How to do it correctly?

### ISSUE THREAD

**Issue by NQUARE** [moses-palmer/pynput#353]
Hello everybody!
I don't understand how to make it possible to control and listen to the mouse.
I'm new to python. Here's all I could put together from the documentation:
```
from pynput.mouse import Button, Controller

mouse = Controller()

movi = False
mx = 0
my = 0

def on_move(x, y):
    if (movi):
    	mouse.position = (mx, my)
    	print(mx)

def on_click(x, y, button, pressed):
    if pressed:
        if button == mouse.Button.left:
            movi = True
            mx = x
            my = y
    else:
        movi = False

with mouse.Listener(on_move = on_move, on_click = on_click) as listener:
    listener.join()
```
But it doesn't work.
Thank you all in advance!

**Comment by moses-palmer** [comment]
One obvious point to start looking is your `on_click` function.

In *Python*, when you assign to a variable (`movi = True`), the runtime must first decide whether you are declaring a new variable or binding an existing variable to a new value.

In your case, you declare `movi`, `mx` and `my` at the top of the script, and then attempt to rebind them in the callbacks. This does not work however, as *Python* requires you to be explicit when you rebind global variables---see [here](https://docs.python.org/3/faq/programming.html#what-are-the-rules-for-local-and-global-variables-in-python).

In short, add the statement `global movi, mx, my` to the top of your function.
